jina/serve/streamer.py

```diff
@@ -224,31 +224,33 @@ def _set_env_streamer_args(**kwargs):
 
 
 class ExecutorStreamer:
-    def __init__(self, gateway_streamer: GatewayStreamer) -> None:
-        self._gateway_streamer = gateway_streamer
-    
+    def __init__(self, connection_pool: GrpcConnectionPool) -> None:
+        self._connection_pool: GrpcConnectionPool = connection_pool
+
     def __getitem__(self, executor_name: str):
         async def _acall_executor(
             docs: DocumentArray,
             request_size: int = 100,
-            return_results: bool = False,
             exec_endpoint: Optional[str] = None,
             parameters: Optional[Dict] = None,
-            results_in_order: bool = False,
+            **kwargs,
         ):
-            result_docs = DocumentArray.empty()
-            async for _docs in self._gateway_streamer.stream_docs(
-                exec_endpoint=exec_endpoint,
-                docs=docs,
-                parameters=parameters,
-                request_size=request_size,
-                return_results=return_results,
-                results_in_order=results_in_order,
-                target_executor=executor_name
-            ):
-                result_docs.extend(_docs)
+            from jina.clients.request import request_generator
```

Comment:
avoid this `request_generator`. Create a Request using the `Request` object. This `generator I would love to remove` since it is too complex

Reply:
ah ok then